viadot/sources/s3.py

```diff
@@ -1,266 +1,132 @@
-        Copies the contents of `from_path` to `to_path`.
-
-        Args:
-            from_path (str, optional): Path from which to copy file(s).
-                Defauls to None.
-            to_path (str, optional): Path where to copy file(s). Defaults
-                to None.
-            recursive (bool, optional): Whether to copy file(s) recursively.
-                Defaults to False.
-        """
-        from_path = from_path or self.path
-        to_path = to_path
-        self.fs.cp(from_path, to_path, recursive=recursive)
+        wr.s3.delete_objects(boto3_session=self.session, path=path)
 
     def from_df(
-        self, df: pd.DataFrame, path: str = None, overwrite: bool = False
-    ) -> None:
+        self,
+        df: pd.DataFrame = None,
+        to_path: str = None,
+        **kwargs,
+    ):
         """
-        Upload a pandas `DataFrame` to a file on Azure Data Lake.
-
+        Upload a pandas `DataFrame` to a csv or parquet file. You can choose different file backends, and have the option of compression.
         Args:
-            df (pd.DataFrame): The pandas `DataFrame` to upload.
-            path (str, optional): The destination path. Defaults to None.
-            overwrite (bool): Whether to overwrite the file if it exist.
+            df (pd.DataFrame, optional): Pandas DataFrame
+            to_path (str, optional): Path to a S3 folder. Defaults to None.
         """
 
-        path = path or self.path
-
-        extension = path.split(".")[-1]
-        if extension not in ("csv", "parquet"):
-            if "." not in path:
-                msg = "Please provide the full path to the file."
-            else:
-                msg = "Accepted file formats are 'csv' and 'parquet'."
-            raise ValueError(msg)
-
-        file_name = path.split("/")[-1]
-        if extension == "csv":
-            # Can do it simply like this if ADLS accesses are set up correctly
-            # url = os.path.join(self.base_url, path)
-            # df.to_csv(url, storage_options=self.storage_options)
-            df.to_csv(file_name, index=False)
+        if to_path.endswith(".csv"):
+            wr.s3.to_csv(
+                boto3_session=self.session,
+                df=df,
+                path=to_path,
+                dataset=True,
+                **kwargs,
+            )
+        elif to_path.endswith(".parquet"):
+            wr.s3.to_parquet(
+                boto3_session=self.session,
+                df=df,
+                path=to_path,
+                dataset=True,
+                **kwargs,
+            )
         else:
-            df.to_parquet(file_name, index=False)
-
-        self.upload(from_path=file_name, to_path=path, overwrite=overwrite)
+            raise ValueError("Only CSV and parquet formats are supported.")
 
-        os.remove(file_name)
+    def to_df(
+        self,
+        from_path: str = None,
+        **kwargs,
+    ):
+        """
+        Reads a csv or parquet file to dataframe.
```

Comment:
FYI, it's good practice to use formatting as it renders better in the docs (as in here eg.: `DataFrame`). Also, pls specify it's a pandas DataFrame, as there are different kinds of dataframes.

Reply:
Updated as per [🎨 updated docstrings and fixed typos](https://github.com/dyvenia/viadot/pull/579/commits/3080223b0461b46ad9260ba51f35b6a41b4a0812)